### PR TITLE
Fixes a bug in combined scripts when js files don't use semicolons

### PR DESCRIPTION
**Pull request by xogg**

### CODE REVIEW

**Comment by nicolasbadia** [comment]
LGTM

File: lib/controllers/combined_files.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ BT.CombinedFilesController = SC.ArrayController.extend({
 
     files.forEach(function (f) {
       ret.push("/* begin of file %@ */".fmt(f.get('path')));
-      ret.push(f.get('content') || "");
+      ret.push((f.get('content') || "") + ';');
```
Comment (mauritslamers) [inline lib/controllers/combined_files.js]: Isn't this going to be problematic when files turn out to be empty? I would think that a check like 
```
var c = (f.get('content') || '').trim();
if (c && c.lastIndexOf(';') !== c.length - 1) c += ';';
ret.push(c);
```
would solve the problem a bit more reliable. Thoughts?

File: lib/controllers/combined_files.js [hunk 2]
```diff
@@ -107,7 +107,8 @@ BT.CombinedFilesController = SC.ArrayController.extend({
 
     files.forEach(function (f) {
       ret.push("/* begin of file %@ */".fmt(f.get('path')));
-      ret.push((f.get('content') || "") + ';');
+      ret.push(f.get('content') || "");
+      ret.push(';');
```
Comment (mauritslamers) [inline lib/controllers/combined_files.js]: See comment on the previous commit.

File: lib/controllers/combined_files.js [hunk 3]
```diff
@@ -108,7 +108,7 @@ BT.CombinedFilesController = SC.ArrayController.extend({
     files.forEach(function (f) {
       ret.push("/* begin of file %@ */".fmt(f.get('path')));
       ret.push(f.get('content') || "");
-      ret.push(';');
+      if(f.isScript) ret.push(';');
```
Comment (mauritslamers) [inline lib/controllers/combined_files.js]: Even this won't really fix what you are trying to do. It still can cause spurious semicolons to appear.

**Comment by mauritslamers** [comment]
It is not entirely clear what the cause of the problem is, except the missing semicolons. In what way is the missing semicolon causing issues?

**Comment by xogg** [comment]
The missing semicolon is not itself a problem at all, because JavaScript permits code without semicolons.

However, a special care should be taken with code without semicolons.
Let's assume, there's a file with the following content:
```
SC.Class.create({
...
})
```
and there's another file defined as:
```
(function()
{
...
})()
```
Obviously, when both files are combined in one, the resulting code will take a created object as a function and will try to call it, passing an anonymous function as a parameter. 
The result of this operation will be called again without parameters.

Also, there's no any problem at all with spurious semicolons, because the code `;;;;;` is an absolutely correct JavaScript code consisting of various empty operators.

So, the original fix is still correct.

**Comment by mauritslamers** [comment]
Ok, clear.

**Pull request merged**